tests/unit/kms.js

```diff
@@ -0,0 +1,181 @@
+import assert from 'assert';
+import kms from '../../lib/kms/wrapper';
+import { cleanup, DummyRequestLogger } from './helpers';
+
+const log = new DummyRequestLogger();
+
+describe('KMS unit tests', () => {
+    beforeEach(() => {
+        cleanup();
+    });
+
+    it('should construct a sse info object on AES256', done => {
+        const algorithm = 'AES256';
+        const headers = {
+            'x-amz-scal-server-side-encryption': algorithm,
+        };
+        kms.bucketLevelEncryption(
+            'dummyBucket', headers, log,
+            (err, sseInfo) => {
+                assert.strictEqual(err, null);
+                assert.strictEqual(sseInfo.cryptoScheme, 1);
+                assert.strictEqual(sseInfo.mandatory, true);
+                assert.strictEqual(sseInfo.algorithm, algorithm);
+                assert.notEqual(sseInfo.masterKeyId, undefined);
+                assert.notEqual(sseInfo.masterKeyId, null);
+                done();
+            });
+    });
+
+    it('should construct a sse info object on aws:kms', done => {
+        const algorithm = 'aws:kms';
+        const masterKeyId = 'foobarbaz';
+        const headers = {
+            'x-amz-scal-server-side-encryption': algorithm,
+            'x-amz-scal-server-side-encryption-aws-kms-key-id': masterKeyId,
+        };
+        kms.bucketLevelEncryption(
+            'dummyBucket', headers, log,
+            (err, sseInfo) => {
+                assert.strictEqual(err, null);
+                assert.strictEqual(sseInfo.cryptoScheme, 1);
+                assert.strictEqual(sseInfo.mandatory, true);
+                assert.strictEqual(sseInfo.algorithm, 'aws:kms');
+                assert.strictEqual(sseInfo.masterKeyId, masterKeyId);
+                done();
+            });
+    });
+
+    it('should not construct a sse info object on garbage', done => {
```

Comment:
'should not construct a sse info object if x-amz-scal-server-side-encryption header contains invalid algorithm option'